.github/guidelines/CODING_GUIDELINES.md

```diff
@@ -0,0 +1,77 @@
+# MetaMask Coding Guidelines
+
+### 1. New Code Should be Typescript
+- New components and utilities should be written in Typescript and enforce typing.
+- Existing code should be refactored into Typescript where time allows. If you are replacing a component, use Typescript.
+
+### 2. Using Functional Components and Hooks Instead of Classes
+- Use functional components and hooks as they result in more concise and readable code compared to classes.
+
+### 3. Organize Files Related to the Same Component in One Folder
+- An example of a component file structure:
+
+```.tsx
+AvatarAccount
+├── AvatarAccount.constants.ts
+├── AvatarAccount.stories.tsx
+├── AvatarAccount.styles.ts
+├── AvatarAccount.test.tsx
+├── AvatarAccount.tsx
+├── AvatarAccount.types.ts
+├── README.md
+├── __snapshots__
+│   └── AvatarAccount.test.tsx.snap
+└── index.ts
```

Comment:
I don't think there is any reason why we couldn't consolidate file name casing here but the current convention is kebab-case. I think this is the same as @DDDDDanica suggestion e.g.
```suggestion
avatar-account
├── avatar-account.stories.tsx
├── avatar-account.scss
├── avatar-account.test.tsx
├── avatar-account.tsx
├── avatar-account.types.ts
├── README.md
├── __snapshots__
│   └── avatar-account.test.tsx.snap
└── index.ts
```

Reply:
Fixed in this [commit](https://github.com/MetaMask/metamask-extension/pull/20651/commits/4b5fdc9c3ee5acfc767c7ca5668badec31c58cb9).